### PR TITLE
Support robust buffer access for drawArrays.

### PR DESCRIPTION
**Pull request by null77**
Currently out-of-bounds drawArrays is tested to verify that the implementation
returns INVALID_OPERATION. The WebGL spec also allows for robust buffer access
when accessing array buffers out-of-bounds. WebGL already supports robust
access behaviour for drawElements. Because there are some significant perf
advantages to robust buffer access for drawArrays as wel this is worth
supporting.

See https://www.khronos.org/registry/webgl/specs/latest/1.0/#4.5
Also see http://anglebug.com/3000

A few WebGL 1.0 test changes are neccessary:

 * rendering/out-of-bounds-array-buffers: test added.
   * Verifies new behaviour much in the same way as the indexed test.
   * Mirrors out-of-bounds-index-buffers test for drawElements.
 * js/tests/out-of-bounds-test: accept NO_ERROR.
   * Fixes potential TDR with huge size and accepts NO_ERROR.
   * Mirrors f51f2b7120c for drawElements support.
 * more/functions/drawArraysOutOfBounds: test removed.
   * Not relevant when robust access is accepted.
   * Mirrors 714ccda0287 removal of drawElements test.
 * more/functions/drawArrays: removed out of bounds error checks.
   * Not relevant when robust access is accepted.
 * rendering/out-of-bounds-index-buffers: updated message.

Existing implementations shouldn't need any changes to pass the updated tests.

@kenrussell @jdashg  PTAL. Testing locally gave no errors with or without the local ANGLE CL on Windows. I'm trying to test this change on the Chrome infra but am having one issue trying with the new ANGLE behaviour:

https://chromium-review.googlesource.com/c/chromium/src/+/1425859/1 shows a passing run with current ANGLE and updated test. Also note anglebug.com links are temporarily broken, please use https://bugs.chromium.org/p/angleproject/issues/detail?id=3000 . But they will be fixed soon.

### CODE REVIEW

File: sdk/tests/conformance/rendering/out-of-bounds-array-buffers.html [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        wtu.checkCanvas(gl, [0, 0, 255, 255]);
+    } else if (error === gl.NO_ERROR) {
+        testPassed("drawArrays flagged NO_ERROR, which is valid so long as all canvas pixels are green.");
+        wtu.checkCanvas(gl, [0, 255, 0, 255]);
+    } else {
+        testFailed("Invalid error flagged by drawArrays. Should be INVALID_OPERATION or NO_ERROR");
+    }
+}
+
+// Create a vertex buffer with 200 properly formed tri-strip quads. These quads will cover the
```
Comment (jdashg) [inline sdk/tests/conformance/rendering/out-of-bounds-array-buffers.html]: This VBO is triangles, not triangle_strip. (It's also CW-winding, which is atypical, but CULL_FACE is disabled by default, so this should still be ok)

File: sdk/tests/conformance/rendering/out-of-bounds-array-buffers.html [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+// Draw out-of-bounds beginning with the start offset passed in.
+// Ensure that drawArrays flags either no error or INVALID_OPERATION. In the case of INVALID_OPERATION,
+// no canvas pixels can be touched.  In the case of NO_ERROR, all written values must either be the
+// zero vertex or a value in the vertex buffer.  See vsCheckOutOfBounds shader.
+function drawAndVerifyOutOfBoundsArrays(gl, first, count) {
+    gl.clearColor(0.0, 0.0, 1.0, 1.0);  // Start with blue to indicate no pixels touched.
+    gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT);
+
+    gl.drawArrays(gl.TRIANGLE_STRIP, first, count);
```
Comment (jdashg) [inline sdk/tests/conformance/rendering/out-of-bounds-array-buffers.html]: `TRIANGLES`

File: sdk/tests/conformance/rendering/out-of-bounds-array-buffers.html (hunk 1 above)
Comment (null77) [reply sdk/tests/conformance/rendering/out-of-bounds-array-buffers.html]: Done, good catch.

File: sdk/tests/conformance/rendering/out-of-bounds-array-buffers.html (hunk 2 above)
Comment (null77) [reply sdk/tests/conformance/rendering/out-of-bounds-array-buffers.html]: Done.

**Comment by jdashg** [comment]
Thanks!

**Comment by null77** [comment]
Thanks to you as well.

**Pull request merged**